internal/container/default_registry.go

```diff
@@ -0,0 +1,31 @@
+package container
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/docker/distribution/reference"
+)
+
+var escapeRegex = regexp.MustCompile(`[/:@]`)
+
+func escapeName(s string) string {
+	return string(escapeRegex.ReplaceAll([]byte(s), []byte("_")))
+}
+
+// Produces a new image name that is in the specified registry.
+// The name might be ugly, favoring uniqueness and simplicity and assuming that the prettiness of ephemeral dev image
+// names is not that important.
+func ReplaceRegistry(defaultRegistry string, rs RefSelector) (RefSelector, error) {
+	if defaultRegistry == "" {
+		return rs, nil
+	}
+
+	newNs := fmt.Sprintf("%s/%s", defaultRegistry, escapeName(rs.RefName()))
```

Comment:
i think this is going to break the case where we have image builds for the same name but different tags? see
https://github.com/jazzdan/tilt_debug
for an example

Reply:
as discussed offline, image builds for same name but different tags have two potential problems:
1. at tiltfile assembly time, joining image builds with k8s resources
2. when monitoring pods, finding the container for a k8s resource

The return value of this is not used at all for (1), and (2) is the problem kinda addressed by Maia's branch, and which behavior is already buggy, and I don't think this is changing.

I've also run this branch against Dan's tilt debug repo and it worked fine, and added a new unit test (`TestDefaultRegistryTwoImagesOnlyDifferByTag`) to verify the behavior. LMK if I've misunderstood the problem.